core/trino-main/src/main/java/io/trino/sql/planner/planprinter/HashCollisionPlanNodeStats.java

```diff
@@ -92,7 +92,6 @@ private static double computedWeightedStdDev(double sumSquared, double sum, doub
     @Override
     public PlanNodeStats mergeWith(PlanNodeStats other)
     {
-        checkMergeable(other);
```

Comment:
When `this` is  not compatible with `other`, we should ensure the end results doesn't depend on ordering of these merges. in particular, we shouldn't probably return `HashCollisionPlanNodeStats` below, since `other.mergeWith(this)` wouldn't do that.